test/miner_txn_mgr_SUITE.erl

```diff
@@ -358,7 +366,433 @@ txn_dependent_test(Config) ->
+
+    %% query the txn mgr txn status API
+    %% queue data will not be available and will default to zero
+    ok = miner_ct_utils:wait_for_gte(height_exactly, Miners, Height + 3),
+    {ok, Txn1Key, _, Txn1SubmitHeight, [{ConMiner1Addr, _, 0, 0}]} =
+        txn_mgr_query_txn(Txn1Key, Miner),
+    {ok, Txn2Key, _, Txn2SubmitHeight, [{ConMiner1Addr, _, 0, 0}]} =
+        txn_mgr_query_txn(Txn2Key, Miner),
+    {ok, Txn3Key, _, Txn3SubmitHeight, [{ConMiner1Addr, _, 0, 0}]} =
+        txn_mgr_query_txn(Txn3Key, Miner),
+    {ok, Txn4Key, _, Txn4SubmitHeight, [{ConMiner1Addr, _, 0, 0}]} =
+        txn_mgr_query_txn(Txn4Key, Miner),
+
+    ok.
+
+txn_queue_common_test(Config) ->
+    %% this sets up and runs the elements of the
+    %% txn queue tests which are common
+    %% across all 3 protocol versions
+    %% any data required by the parent
+    %% will be added to the config
+
+    Miners = ?config(miners, Config),
+    NonConMiners = ?config(non_consensus_miners, Config),
+    Miner = hd(NonConMiners),
+    ConMiners = ?config(consensus_miners, Config),
+    ConMiner1 = hd(ConMiners),
+    AddrList = ?config(tagged_miner_addresses, Config),
+
+    Addr = miner_ct_utils:node2addr(Miner, AddrList),
+    ConMiner1Addr = miner_ct_utils:node2addr(ConMiner1, AddrList),
+
+    Chain = ct_rpc:call(Miner, blockchain_worker, blockchain, []),
+    ct:pal("consensus miners ~p", [ConMiners]),
+    ct:pal("non consensus miners ~p", [NonConMiners]),
+    ct:pal("miner in use ~p", [Miner]),
+    ct:pal("consensus miner in use ~p", [ConMiner1]),
+    ct:pal("consensus miner bin in use ~p", [ConMiner1Addr]),
+
+    PayerAddr = Addr,
+    Payee = hd(miner_ct_utils:shuffle(ConMiners)),
+    PayeeAddr = miner_ct_utils:node2addr(Payee, AddrList),
+    {ok, _Pubkey, SigFun, _ECDHFun} = ct_rpc:call(Miner, blockchain_swarm, keys, []),
+    IgnoredTxns = [],
+
+    %% meck out the sidecar random_n function
+    %% make it return an empty list
+    %% simulate an empty buffer
+    %% this should result in the submitted txns
+    %% remaining in the sidecar buffer
+    meck:new(hbbft_utils, [passthrough]),
+    [
+            ok =
+            ct_rpc:call(
+                Node,
+                meck,
+                expect,
+                [hbbft_utils, random_n, fun(_, _) -> [] end],
+                3000
+            )
+    ||
+        Node <- ConMiners
+    ],
+
+    %% meck out the signatory_rand_members function
+    %% force it to always return a deterministic list of acceptors
+    %% which in this case will be a single consensus member
+    %% this will result in the txn mgr pushing its cached txns
+    %% to this single CG member
+    %% and thus we can ensure we only need to check the sidecar
+    %% buffer on a single and known CG member
+
+    %% also meck out is_valid for payment txns
+    %% this test is not concerned with validity issues
+    meck:new(blockchain_txn_mgr, [passthrough]),
+    meck:new(blockchain_txn_payment_v1, [passthrough]),
```

Comment:
any reason to suspect a v2 payment wouldn't work just as well here for validation purposes if v1 is deprecated (it is, isn't it?) other than creating needing to create the payments separately from the txn?

Reply:
any type of txn with a nonce check dependency will work here.  v1 payments were used as that is what was already in the existing mgr tests